src/sidebar/services/groups.js

```diff
@@ -163,164 +163,144 @@ function groups(
-            // Set the directLinkedAnnotationGroupId to be used later in
-            // the filterGroups method.
-            directLinkedAnnotationGroupId = directLinkedAnn.group;
-
-            const directLinkedAnnGroup = myGroups
-              .concat(allFeaturedGroups)
-              .some(g => g.id === directLinkedAnn.group);
-
-            // If the direct-linked annotation's group has not already been fetched,
-            // fetch it.
-            if (!directLinkedAnnGroup) {
-              const initialFeaturedGroups = allFeaturedGroups;
-              allFeaturedGroups = fetchGroup({
-                id: directLinkedAnn.group,
-                expand: params.expand,
-              }).then(directLinkedAnnGroup => {
-                if (!directLinkedAnnGroup) {
-                  return initialFeaturedGroups;
-                }
-                return initialFeaturedGroups.concat(directLinkedAnnGroup);
-              });
-            }
-          }
-          return Promise.all([myGroups, allFeaturedGroups, documentUri, token]);
-        }
-      )
-      .then(([myGroups, featuredGroups, documentUri, token]) => {
-        // Step 4. Combine all the groups into a single list and set additional
-        // metadata on them that will be used elsewhere in the app.
-        const isLoggedIn = token !== null;
-        const groups = filterGroups(
-          combineGroups(myGroups, featuredGroups, documentUri),
-          isLoggedIn,
-          directLinkedAnnotationGroupId,
-          directLinkedGroupId
-        );
-
-        injectOrganizations(groups);
-
-        // Step 5. Load the groups into the store and focus the appropriate
-        // group.
-        const isFirstLoad = store.allGroups().length === 0;
-        const prevFocusedGroup = localStorage.getItem(STORAGE_KEY);
-
-        store.loadGroups(groups);
-
-        if (
-          isFirstLoad &&
-          groups.some(g => g.id === directLinkedAnnotationGroupId)
-        ) {
-          store.focusGroup(directLinkedAnnotationGroupId);
-        } else if (
-          isFirstLoad &&
-          groups.some(g => g.id === directLinkedGroupId)
-        ) {
-          store.focusGroup(directLinkedGroupId);
-        } else if (isFirstLoad && groups.some(g => g.id === prevFocusedGroup)) {
-          store.focusGroup(prevFocusedGroup);
+        if (directLinkedAnnGroup) {
+          featuredGroups.push(directLinkedAnnGroup);
         }
+      }
+    }
 
-        return groups;
-      });
+    // Step 4. Combine all the groups into a single list and set additional
+    // metadata on them that will be used elsewhere in the app.
+    const isLoggedIn = token !== null;
+    const groups = filterGroups(
+      combineGroups(myGroups, featuredGroups, documentUri),
+      isLoggedIn,
+      directLinkedAnnotationGroupId,
+      directLinkedGroupId
+    );
+
+    injectOrganizations(groups);
+
+    // Step 5. Load the groups into the store and focus the appropriate
```

Comment:
The step comments help break up the problem. Just thinking out loud here but, they might be good markers for refactoring these into functions later.

Reply:
Perhaps. The goal of this PR was to make the `load` function as "direct" as possible, ie. a set of sequential steps with minimal branching.

Reply:
Just a thought certainly not something I intended to be done for this particular PR. :)